cpp/oneapi/dal/backend/primitives/lapack/solve_dpc.cpp

```diff
@@ -70,12 +233,34 @@ sycl::event solve_system(sycl::queue& queue,
 
     auto [nxty, xty_event] = copy<ndorder::c, Float, ylayout, alloc>(queue, xty, dependencies);
     auto [nxtx, xtx_event] = copy<ndorder::c, Float, xlayout, alloc>(queue, xtx, dependencies);
+    const std::int64_t dim_xtx = xtx.get_dimension(0);
+
+    sycl::event solution_event;
 
     opt_array<Float> dummy{};
-    auto potrf_event = potrf_factorization<uplo>(queue, nxtx, dummy, { xtx_event });
-    auto potrs_event = potrs_solution<uplo>(queue, nxtx, nxty, dummy, { potrf_event, xty_event });
+    try {
```

Comment:
As I remember we use try catch only in tests/examples. Does it make sense to avoid it here?

Reply:
It's meant to catch an exception thrown by oneMKL in the call to `potrf`, over which we have no control AFAIK. An alternative would be to modify the wrapper, but it's also used elsewhere in places where the exception is not catched (so a user error is thrown).

Reply:
I am Ok with having a `try` in the code; but let's make it more clear for the readers - please add the description of what is happening here - why the system is tries to be solved twice in some cases.

Reply:
Added a comment about the logic for the try-catch block.